zircon.core/src/commonMain/kotlin/org/hexworks/zircon/api/fragment/Table.kt

```diff
@@ -0,0 +1,31 @@
+package org.hexworks.zircon.api.fragment
+
+import org.hexworks.cobalt.databinding.api.value.ObservableValue
+import org.hexworks.zircon.api.component.Fragment
+import org.hexworks.zircon.api.data.Size
+
+/**
+ * A table fragment displays data in rows. Each row contains several cells. How a cell is displayed depends
+ * on the definition of it's column.
+ *
+ * A table contains data of type [M] (the model). Every model object represents one row in the table. **A table
+ * may not be empty**.
+ *
+ * @param M the type of the underlying model representing this table's data.
+ */
+interface Table<M : Any> : Fragment {
+    /**
+     * The element representing the currently selected row.
+     */
+    val selectedRow: M
+
+    /**
+     * The element representing the currently selected row as [ObservableValue].
+     */
+    val selectedRowValue: ObservableValue<M>
```

Comment:
You can kill two birds with one stone and modify this to be an `ObservableList` and name it `selectedRowsValue`. Then an empty list means no selection and you can also have multiple selected rows. You can change `selectedRow` accordingly.